app/frontend/src/features/communities/CommunityPage/CommunityPage.tsx

```diff
@@ -0,0 +1,445 @@
+    },
+  },
+  breadcrumbs: {
+    "& ol": {
+      justifyContent: "center",
+      [theme.breakpoints.up("md")]: {
+        justifyContent: "flex-start",
+      },
+    },
+  },
+  title: {
+    marginTop: theme.spacing(1),
+    marginBottom: theme.spacing(1),
+  },
+  description: {
+    marginBottom: theme.spacing(1),
+  },
+  navButtonContainer: {
+    display: "flex",
+    justifyContent: "space-around",
+    marginBottom: theme.spacing(1),
+    [theme.breakpoints.only("sm")]: {
+      justifyContent: "center",
+      "& > * + *": {
+        marginInlineStart: theme.spacing(4),
+      },
+    },
+    [theme.breakpoints.up("md")]: {
+      width: "30%",
+    },
+  },
+  cardContainer: {
+    [theme.breakpoints.down("xs")]: {
+      //break out of page padding
+      width: "100vw",
+      position: "relative",
+      left: "50%",
+      right: "50%",
+      marginLeft: "-50vw",
+      marginRight: "-50vw",
+    },
+    [theme.breakpoints.up("sm")]: {
+      marginTop: theme.spacing(1),
+      marginBottom: theme.spacing(2),
+      display: "flex",
+      flexDirection: "row",
+      flexWrap: "wrap",
+      justifyContent: "space-between",
+      "&::after": {
+        content: "''",
+        flexBasis: "100%",
+        [theme.breakpoints.up("sm")]: {
+          flexBasis: `calc(50% - ${theme.spacing(1)})`,
+        },
+        [theme.breakpoints.up("md")]: {
+          flexBasis: `calc(33.33% - ${theme.spacing(1)})`,
+        },
+      },
+    },
+  },
+  loadMoreButton: {
+    alignSelf: "center",
+    [theme.breakpoints.up("sm")]: {
+      width: `calc(50% - ${theme.spacing(1)})`,
+    },
+    [theme.breakpoints.up("md")]: {
+      width: `calc(33% - ${theme.spacing(1)})`,
+    },
+  },
+  placeEventCard: {
+    marginBottom: theme.spacing(1),
+    width: 200,
+    [theme.breakpoints.up("sm")]: {
+      width: `calc(50% - ${theme.spacing(1)})`,
+    },
+    [theme.breakpoints.up("md")]: {
+      width: `calc(33% - ${theme.spacing(1)})`,
+    },
+  },
+  discussionsHeader: {
+    display: "flex",
+    justifyContent: "space-between",
+    alignItems: "center",
+  },
+  newPostButton: {
+    margin: theme.spacing(1),
+  },
+  discussionsContainer: {
+    display: "flex",
+    flexDirection: "row",
+    flexWrap: "wrap",
+    justifyContent: "space-between",
+    "& > *": {
+      width: "100%",
+      [theme.breakpoints.up("sm")]: {
+        width: `calc(50% - ${theme.spacing(1)})`,
+      },
+      [theme.breakpoints.up("md")]: {
+        width: `calc(33.33% - ${theme.spacing(1)})`,
+      },
+    },
+    //preserve grid in the last row
+    "&::after": {
+      content: "''",
+      flexBasis: "100%",
+      [theme.breakpoints.up("sm")]: {
+        flexBasis: `calc(50% - ${theme.spacing(1)})`,
+      },
+      [theme.breakpoints.up("md")]: {
+        flexBasis: `calc(33.33% - ${theme.spacing(1)})`,
+      },
+    },
+  },
+  discussionCard: {
+    marginBottom: theme.spacing(1),
+  },
+}));
+
+export default function CommunityPage() {
```

Comment:
This is still a quite a big component... Could the places and discussions not be broken up further into its own with the relevant hooks moving in with them?